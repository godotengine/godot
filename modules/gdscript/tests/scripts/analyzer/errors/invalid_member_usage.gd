# In 3.x there are warnings `CONSTANT_USED_AS_FUNCTION`, `FUNCTION_USED_AS_PROPERTY`, and `PROPERTY_USED_AS_FUNCTION`.
# In 4.x they are deprecated because either an error is produced or a `Callable` is returned.

const CONSTANT = 25

var property = 25

func function():
	pass

func test():
	function = 25
	CONSTANT(123)
	property()
