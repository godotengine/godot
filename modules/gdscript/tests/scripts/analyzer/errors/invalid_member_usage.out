GDTEST_ANALYZER_ERROR
>> ERROR at line 12: Cannot assign a new value to a constant.
>> ERROR at line 13: Member "CONSTANT" is not a function.
>> ERROR at line 13: Name "CONSTANT" called as a function but is a "int".
>> ERROR at line 14: Member "property" is not a function.
>> ERROR at line 14: Name "property" called as a function but is a "int".
